akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala

```diff
@@ -701,7 +749,88 @@ class EventSourcedBehaviorSpec extends ScalaTestWithActorTestKit(EventSourcedBeh
         c2 ! Fail
         probe.expectTerminated(c2) // should fail
       }
+    }
+
+    "delete snapshots automatically, based on criteria" in {
+      val unexpected = (signal: EventSourcedSignal) => fail(s"Unexpected signal [$signal].")
+
+      val pid = nextPid
+      val snapshotProbe = TestProbe[Try[Done]]()
+      val retentionProbe = TestProbe[Try[EventSourcedSignal]]()
+      val replyProbe = TestProbe[State]()
+
+      val persistentActor = spawn(
+        Behaviors.setup[Command](
+          ctx ⇒
+            counterWithSnapshotAndRetentionProbe(ctx, pid, snapshotProbe.ref, retentionProbe.ref)
+              .snapshotEvery(2)
+              .withRetention(
+                RetentionCriteria(snapshotEveryNEvents = 2, keepNSnapshots = 2, deleteEventsOnSnapshot = false))))
```

Comment:
Created separate issue https://github.com/akka/akka/issues/26544 for `.snapshotEvery(2)` and `snapshotEveryNEvents = 2` redundancy. Followup in separate PR.